core/dnsserver/server.go

```diff
@@ -81,6 +87,17 @@ func NewServer(addr string, group []*Config) (*Server, error) {
 		// append the config to the zone's configs
 		s.zones[site.Zone] = append(s.zones[site.Zone], site)
 
+		// set timeouts
+		if site.ReadTimeout != time.Duration(0) {
+			s.readTimeout = site.ReadTimeout
+		}
+		if site.WriteTimeout != time.Duration(0) {
+			s.writeTimeout = site.WriteTimeout
+		}
+		if site.IdleTimeout != time.Duration(0) {
+			s.idleTimeout = site.IdleTimeout
+		}
+
 		// copy tsig secrets
```

Comment:
Wondering if there is a better way to do this (multiple if)

Reply:
Maybe?  But it looks perfectly fine and easy to read as is.